x/incentive/types/msg_test.go

```diff
@@ -0,0 +1,59 @@
+package types_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/incentive/types"
+	"github.com/stretchr/testify/suite"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+type msgTest struct {
+	from       sdk.AccAddress
+	denom      string
+	expectPass bool
+}
+
+type MsgTestSuite struct {
+	suite.Suite
+
+	tests []msgTest
+}
+
+func (suite *MsgTestSuite) SetupTest() {
+	tests := []msgTest{
+		msgTest{
+			from:       sdk.AccAddress(crypto.AddressHash([]byte("KavaTest1"))),
+			denom:      "bnb",
+			expectPass: true,
+		},
+		msgTest{
+			from:       sdk.AccAddress(crypto.AddressHash([]byte("KavaTest1"))),
+			denom:      "",
+			expectPass: false,
+		},
+		msgTest{
+			from:       sdk.AccAddress{},
+			denom:      "bnb",
+			expectPass: false,
+		},
+	}
+	suite.tests = tests
+}
+
+func (suite *MsgTestSuite) TestMsgValidation() {
+	for _, t := range suite.tests {
+		msg := types.NewMsgClaimReward(t.from, t.denom)
+		err := msg.ValidateBasic()
+		if t.expectPass {
+			suite.NoError(err)
```

Comment:
suggestion for the future, use `suite.Require().NoError(err)` since the default for the testing suite is assert

cc @karzak

Reply:
Updated on #429